src/main/java/edu/hm/hafner/coverage/parser/CoberturaParser.java

```diff
@@ -140,17 +141,20 @@ else if (event.isEndElement()) {
 
     private FileNode createFileNode(final StartElement element, final PackageNode packageNode) {
         var fileName = getValueOf(element, FILE_NAME);
-        var path = getTreeStringBuilder().intern(PATH_UTIL.getRelativePath(fileName));
-
-        return packageNode.findOrCreateFileNode(getFileName(fileName), path);
+        var relativePath = PATH_UTIL.getRelativePath(fileName);
+        var actualPath = relativePath.startsWith("/_/") ? 
+                         relativePath.replace("/_/", "./") : 
+                         relativePath;
+        var finalPath = getTreeStringBuilder().intern(actualPath);
+        return packageNode.findOrCreateFileNode(getFileName(fileName), finalPath);
     }
 
     private String getFileName(final String relativePath) {
-        var path = Paths.get(PATH_UTIL.getAbsolutePath(relativePath)).getFileName();
-        if (path == null) {
-            return relativePath;
-        }
-        return path.toString();
+        var fileName = Paths.get(PATH_UTIL.getAbsolutePath(relativePath)).getFileName();
+        var actualFileName = (fileName != null && fileName.toString().startsWith("/_/")) ? 
+                              fileName.toString().replace("/_/", "./") : 
+                              (fileName != null ? fileName.toString() : relativePath);
+        return actualFileName;
```

Comment:
Please extract the file name logic into a method and reuse it here.

Reply:
👍